src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.ts

```diff
@@ -744,4 +805,17 @@ export class TableComponent implements AfterContentChecked, OnInit, OnChanges, O
       };
     };
   }
+
+  toggleExpandRow(row: any, isExpanded: boolean) {
+    if (!isExpanded) {
+      // If current row isn't expanded, collapse others
+      this.table.rowDetail.collapseAllRows();
+      this.setExpandedRow.emit(row);
+    } else {
+      // If all rows are closed, emit undefined
+      this.setExpandedRow.emit(undefined);
+    }
+    this.expanded = row; // TODO improve alles hier sollte nur gesetzt werden wenn wirklich getoggled
```

Comment:
Please translate the TODO or remove or improve it ;)

Reply:
I'll just fix and remove it